utils.go

```diff
@@ -1,65 +1,206 @@
+		viewErrorAndExit("Unable to create the language file.", err.Error())
 		return err
+	} else {
+		// close file
+		defer languageFile.Close()
+		// write the template into the new file
+		_, err = io.Copy(languageFile, strings.NewReader(translationLanguage))
+		if err != nil {
+			viewErrorAndExit("Unable to write into language file.", err.Error())
+			return err
+		} else {
+			addNewLinesToDebug("Language file successfully created.")
+			return nil
+		}
 	}
-	// close file
-	defer languageFile.Close()
-	// write the template into the new file
-	_, err = io.Copy(languageFile, strings.NewReader(translationLanguage))
-	if err != nil {
-		walk.MsgBox(windowMsgBox, "Error", "Unable to write into language file, please contact your local support.", walk.MsgBoxOK)
-		log.Fatal("Failed writing language constant to file: ", err)
-		return err
+}
+
+func addNewLinesToDebug(mytxt string) {
+	if debug {
+		log.Print(mytxt)
+		myTxtC := chunks(mytxt, 60)
+		myCt := debugTxt.Text()
+		debugTxt.SetText(myCt + myTxtC + "\r\n")
+		numlines := strings.Count(debugTxt.Text(), "\r\n")
+		debugHeight := (13 * numlines) + 13
+		if debugHeight <= 240 {
+			debugHeight = 240
+		}
+		debugTxtsize := walk.Size{Width: 400, Height: debugHeight}
+		debugTxt.SetMinMaxSize(walk.Size(debugTxtsize), walk.Size(debugTxtsize))
+	} else {
+		log.Print(mytxt)
 	}
-	log.Print("Language file successfully created.")
-	return nil
 }
 
+// Chunk large string at x charaters
+// Source: https://stackoverflow.com/a/61469854
+func chunks(s string, chunkSize int) string {
+	if len(s) == 0 {
+		return ""
+	}
+	if chunkSize >= len(s) {
+		return s
+	}
+	var chunks []string = make([]string, 0, (len(s)-1)/chunkSize+1)
+	currentLen := 0
+	currentStart := 0
+	for i := range s {
+		if currentLen == chunkSize {
+			chunks = append(chunks, s[currentStart:i])
+			currentLen = 0
+			currentStart = i
+		}
+		currentLen++
+	}
+	chunks = append(chunks, s[currentStart:])
+	justString := strings.Join(chunks, "\r\n")
+	return justString
+}
+
+// Enable the debug view
+func viewDebug() {
+	if debug {
+		debug = false
+		// Change view
+		scb, _ := walk.NewSolidColorBrush(walk.RGB(4, 5, 3))
+		mw1.SetBackground(scb)
+		debugGrpBox.SetVisible(false)
+		imgView.SetVisible(true)
+		viewClosedButton(false)
+		configButton.SetText("Configure")
+	} else {
+		debug = true
+		// Change view
+		scb, _ := walk.NewSolidColorBrush(walk.RGB(255, 255, 255))
+		mw1.SetBackground(scb)
+		debugGrpBox.SetVisible(true)
+		imgView.SetVisible(false)
+		viewClosedButton(true)
+		configButton.SetText("Configure with debug")
+	}
+}
+
+// Enable Close button view
+func viewClosedButton(b bool) {
+	closedButton.SetVisible(b)
+}
+
+// View Error
+func viewErrorAndExit(s string, sExtra string) {
+	if sExtra != "" {
+		addNewLinesToDebug(s + " " + sExtra)
+	}
+	if !debug {
+		walk.MsgBox(windowMsgBox, T("errorWindowTitle"), s+"\r\nPlease enable Debug Mode and contact your local support.", walk.MsgBoxOK)
```

Comment:
Do we keep it or it needs to be replaced ?

Reply:
This one is Yes, we need it because it is the message before exit when something fails. Except if you do not want to see something when the program fails.